hack/metrics_metadata.py

```diff
@@ -0,0 +1,92 @@
+import json
+import sys
+import subprocess
+import time
+import re
+import argparse
+
+
+def parse_args(args):
+    """
+    Parse command line arguments.
+    """
+    parser = argparse.ArgumentParser(
+        description="Get Prometheus metrics metadata and replace the text between <!--METRICS_METADATA_START--> and <!--METRICS_METADATA_END--> with the markdown table.")
+    parser.add_argument("context", help="Kubernetes context")
+    parser.add_argument("namespace", help="Kubernetes namespace")
+    parser.add_argument("file", help="File to update")
+    return parser.parse_args(args)
+
+
+def get_metrics_metadata_from_prometheus(context, namespace):
+    """
+    Function to get Prometheus metrics metadata in JSON format.
+    """
+    try:
+        proc = subprocess.Popen(["kubectl", "--context", context, "--namespace", namespace,
+                                "port-forward", "service/prometheus-" + namespace + "-server", "9090:80"])
+
+        # wait for connection to be established
+        while True:
+            time.sleep(2)
+            try:
+                subprocess.check_output(
+                    ["curl", "--no-progress-meter", "http://localhost:9090/api/v1/metadata"])
```

Comment:
t: you can also make http requests with python (urllib.request or requests lib).